src/components/cylc/cylcObject/Menu.vue

```diff
@@ -305,9 +305,24 @@ export default {
      * @param {Event} e - the click event
      */
     onClickOutside (e) {
-      this.closeMenu()
-      if (e.target?.getAttribute('data-c-interactive')) {
-        this.showMenu = true
+
+      let shouldClose = true
+
+      // check if the thing being clicked on is a child of the thing that the
+      // menu is open for
+      let target = e.target
+      while (target) {
+        if (target?.getAttribute('data-c-interactive')) {
+          // keep the menu open
+          // this.showMenu = true
+          shouldClose = false
+          break
+        }
+        target = target.parentElement
```

Comment:
When we click on a task icon in the graph, the click event will land on render elements, e.g. the outline `<circle />` element within the task group rather than the task `<g />` element itself (the bit which has the `v-cylc-object` directive attached to it.

As a result we can't just look for the `c-interactive` data attribute, we have to walk up the tree as it will belong to a parent element not the thing we clicked on.

This will go away with Ronnie's PR.